pkg/scheduler/backend/api_dispatcher/api_dispatcher.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apidispatcher
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+	fwk "k8s.io/kube-scheduler/framework"
+)
+
+// APIDispatcher implements the fwk.APIDispatcher interface and allows for queueing and dispatching API calls asynchronously.
+type APIDispatcher struct {
+	stop chan struct{}
+
+	client            clientset.Interface
+	callController    *callController
+	goroutinesLimiter *goroutinesLimiter
+}
+
+// New returns a new APIDispatcher object.
+func New(client clientset.Interface, parallelization int, apiCallRelevances fwk.APICallRelevances) *APIDispatcher {
+	d := APIDispatcher{
+		stop:              make(chan struct{}),
+		client:            client,
+		callController:    newCallController(apiCallRelevances),
+		goroutinesLimiter: newGoroutinesLimiter(parallelization),
+	}
+
+	return &d
+}
+
+// Add adds an API call to the dispatcher's queue. It returns an error if the call is not enqueued
+// (e.g., if it's skipped). The caller should handle ErrCallSkipped if returned.
```

Comment:
Maybe similar to the question in another thread, but what would the caller do with this error?